src/relations/db.py

```diff
@@ -189,13 +192,14 @@ def set_up_relation(self, relation: Relation) -> bool:
 
             # Creates the user and the database for this specific relation if it was not already
             # created in a previous relation changed event.
-            user = f"relation-{relation.id}"
             password = unit_relation_databag.get("password", new_password())
 
             # Store the user, password and database name in the secret store to be accessible by
             # non-leader units when the cluster topology changes.
-            self.charm.set_secret(APP_SCOPE, user, password)
-            self.charm.set_secret(APP_SCOPE, f"{user}-database", database)
+            if password != self.charm.get_secret(APP_SCOPE, user):
+                self.charm.set_secret(APP_SCOPE, user, password)
+            if database and database != self.charm.get_secret(APP_SCOPE, f"{user}-database"):
+                self.charm.set_secret(APP_SCOPE, f"{user}-database", database)
```

Comment:
Seems we have some kind of issue when setting multiple secret fields at the same time or when  resetting the same value.